lib/bosh/BoshDirectorClient.js

```diff
@@ -16,27 +16,133 @@ const NotFound = errors.NotFound;
 const BadRequest = errors.BadRequest;
 
 class BoshDirectorClient extends HttpClient {
-  constructor(boshDirectorConfig) {
-    const boshDirector = boshDirectorConfig || config.director;
+  constructor() {
+    // activeConfigs - directors supporting lifecycle oprations
+    let activeConfigs = BoshDirectorClient.getActiveConfigs();
+    // directorConfig - directors supporting 'create' oprations
+    let coreConfigs = BoshDirectorClient.getCoreConfigs(activeConfigs);
+    // inactiveConfigs - directors for OutOfBand backup scenario
+    let inactiveConfigs = BoshDirectorClient.getInactiveConfigs();
+    let config = BoshDirectorClient.getPrimaryConfig(coreConfigs);
```

Comment:
not sure why this is required. A random config is required at the time of creation. No need to create this during initialisation. Just to point out the inconsistency again, here something from PrimaryConfig is being retrieved but in initialization nothing related primary naming convention exists.